Sources/Conformance/conformance.pb.swift

```diff
@@ -147,44 +147,34 @@ struct Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf.Prot
   ]
 
 
-  enum OneOf_Payload: ExpressibleByNilLiteral, SwiftProtobuf.OneofEnum {
+  enum OneOf_Payload: SwiftProtobuf.OneofEnum {
     case protobufPayload(Data)
     case jsonPayload(String)
-    case None
 
     static func ==(lhs: Conformance_ConformanceRequest.OneOf_Payload, rhs: Conformance_ConformanceRequest.OneOf_Payload) -> Bool {
       switch (lhs, rhs) {
       case (.protobufPayload(let l), .protobufPayload(let r)): return l == r
       case (.jsonPayload(let l), .jsonPayload(let r)): return l == r
-      case (.None, .None): return true
       default: return false
       }
     }
 
-    init(nilLiteral: ()) {
-      self = .None
-    }
-
-    init() {
-      self = .None
-    }
-
-    mutating func decodeField<T: SwiftProtobuf.Decoder>(decoder: inout T, fieldNumber: Int) throws {
-      if self != .None {
-        try decoder.handleConflictingOneOf()
-      }
+    init?<T: SwiftProtobuf.Decoder>(byDecodingFrom decoder: inout T, fieldNumber: Int) throws {
       switch fieldNumber {
       case 1:
         var value = Data()
         try decoder.decodeSingularBytesField(value: &value)
         self = .protobufPayload(value)
+        return
       case 2:
         var value = String()
         try decoder.decodeSingularStringField(value: &value)
         self = .jsonPayload(value)
+        return
       default:
-        self = .None
+        break
       }
+      return nil
```

Comment:
Ah, I was looking at a different one (proto2, which had optionals and if-statements). Using a default branch in the cases where conditionals aren't present is tricky because they can be mixed (message vs. scalar fields, for instance).